File: StrategyPattern_C++/BankData.cpp
#include "BankData.h"



BankData::BankData()
{
}


BankData::~BankData()
{
}

void BankData::registerObserver(Observer* o)
{
	observers.push_back(o);
}

void BankData::remove(Observer* o)
{
	observers.clear();
}

void BankData::notifyObserver()
{
	for (Observer* o : observers)
	{
		o->update(msg);
	}
}

void BankData::setMsg(string msg)
{
	this->msg = msg;
	msgChanged();
}

void BankData::msgChanged()
{
	notifyObserver();
}

File: Strategy/Java/bank/BankData.java
package bank;

import java.util.ArrayList;

public class BankData implements Subject
{
	private ArrayList<Observer> observers;
	private String msg;
	
	public BankData()
	{
		observers = new ArrayList<Observer>();
	}
	
	public void register(Observer o)
	{
		observers.add(o);
	}

	public void remove(Observer o) 
	{
		observers.remove(o);
	}

	public void notifyObserver() 
	{
		for (Observer o : observers)
		{
			o.update(msg);
		}
	}
	
	public void setMsg(String msg)
	{
		this.msg = msg;
		msgChanged();
	}
	
	public void msgChanged()
	{
		notifyObserver();
	}
}
